src/pages/requests/users/external-from-main/[requestId]/index.tsx

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+
+import { ExternalFromMainOrgUserCreationRequestPage } from '../../../../../components/ExternalFromMainOrgUserCreationRequestPage/ExternalFromMainOrgUserCreationRequestPage';
+import { pages } from '../../../../../hooks/useRouter';
+import { trpc } from '../../../../../trpc/trpcClient';
+import { createGetServerSideProps } from '../../../../../utils/createGetSSRProps';
+import { InferServerSideProps } from '../../../../../utils/types';
+
+export const getServerSideProps = createGetServerSideProps({
+    requireSession: true,
+    stringIds: { requestId: true },
+    action: async ({ stringIds, session, ssg }) => {
+        if (!session.user.role?.createUser || !session.user.role.editUserCreationRequests) {
+            return {
+                redirect: {
+                    destination: pages.home,
+                    permanent: false,
+                },
+            };
+        }
+        await ssg.userCreationRequest.getById.fetch(stringIds.requestId);
+
+        return { requestId: stringIds.requestId };
+    },
+});
+
+export default function SectionCreationPage({ stringIds }: InferServerSideProps<typeof getServerSideProps>) {
```

Comment:
...but get it from stringIds.requestId
why not just requestId?

Reply:
because requestId returns under condition in getServerSideProps, so it can be undefined